src/NUnitFramework/tests/Internal/Filters/NotFilterTests.cs

```diff
@@ -54,7 +54,7 @@ public void PassTest()
         {
             var filter = new NotFilter(new CategoryFilter("Dummy"));
 
-            Assert.False(filter.Pass(_topLevelSuite));
+            Assert.True(filter.Pass(_topLevelSuite));
             Assert.False(filter.Pass(_dummyFixture));
             Assert.False(filter.Pass(_dummyFixture.Tests[0]));
```

Comment:
Since CompositeWorkItem does not differentiate between Fixtures and Tests, I am not sure it makes any difference, but it is easy enough to add another Test to one of the Fixtures, or to make another Fixture with no category, and put it on one of the Test instead, is that is what you are looking for?

 

From: CharliePoole [mailto:notifications@github.com] 
Sent: Saturday, October 03, 2015 9:11 PM
To: nunit/nunit
Cc: Neil Colvin
Subject: Re: [nunit] Fix NotFilter so -exclude works correctly (#862)

 

In src/NUnitFramework/tests/Internal/Filters/NotFilterTests.cs <https://github.com/nunit/nunit/pull/862#discussion_r41093346> :

> @@ -54,7 +54,7 @@ public void PassTest()
>          {
>              var filter = new NotFilter(new CategoryFilter("Dummy"));
>  
> -            Assert.False(filter.Pass(_topLevelSuite));
> +            Assert.True(filter.Pass(_topLevelSuite));
>              Assert.False(filter.Pass(_dummyFixture));
>              Assert.False(filter.Pass(_dummyFixture.Tests[0]));

There are no fixtures with two tests in them, as far as I can see. That's the exact case that was reported as not working.

—
Reply to this email directly or view it on GitHub <https://github.com/nunit/nunit/pull/862/files#r41093346> .  <https://github.com/notifications/beacon/AFh64cQvcdn9i0RB8BwLcN_CE1O7AM57ks5o4HQzgaJpZM4GIcgJ.gif>